Scripts/DCS-BIOS/lib/modules/aircraft_modules/MiG-29A.lua

```diff
@@ -190,37 +192,84 @@ function MiG_29A:defineCabinTempSwitch(identifier, device_id, arg_number, catego
 	end)
 end
 
---- Adds a new push button control that sends a negative input
---- @param identifier string the unique identifier for the control
---- @param device_id integer the dcs device id
---- @param command integer the dcs command
---- @param arg_number integer the dcs argument number
---- @param category string the category in which the control should appear
---- @param description string additional information about the control
---- @return Control control the control which was added to the module
-function MiG_29A:defineInvertedPushButton(identifier, device_id, command, arg_number, category, description)
-	local control = self:defineTumb(identifier, device_id, command, arg_number, 1, { -1, 0 }, nil, false, category, description)
-	control.api_variant = ApiVariant.momentary_last_position
+local function gunTriggerIntValue(arg_value)
+	if arg_value < 0.30 then
+		return 0
+	elseif arg_value < 0.675 then
+		return 1
+	else
+		return 2
+	end
+end
+
+function MiG_29A:defineGunTrigger(identifier, device_id, arg_number, category, description)
+	local alloc = self:allocateInt(2)
+	self:addExportHook(function(dev0)
+		local val = gunTriggerIntValue(dev0:get_argument_value(arg_number))
+		alloc:setValue(val)
+	end)
+
+	local control = Control:new(category, ControlType.toggle_switch, identifier, description, {
```

Comment:
I can't remember, do we use `toggle_switch` for 3-pos switches?

Reply:
No it's seems it's selector, will change!